Fix the `zebra_scan` crate to work with the last ECC dependencies

In https://github.com/ZcashFoundation/zebra/pull/8810 we upgraded the ECC dependencies to the last version, for Nu6. However, we had to make an exception for the `zebra-scan` crate due to the following building error:

```
error[E0308]: `?` operator has incompatible types
   --> zebra-scan/src/service/scan_task/scan.rs:552:35
    |
552 |       UnifiedFullViewingKey::parse(&Ufvk::try_from_items(vec![Fvk::try_from((
    |  ___________________________________^
553 | |         2,
554 | |         &dfvk.to_bytes()[..],
555 | |     ))?])?)
    | |__________^ expected `zcash_address::kind::unified::fvk::Ufvk`, found `Ufvk`
    |
    = note: `?` operator cannot convert from `Ufvk` to `zcash_address::kind::unified::fvk::Ufvk`
    = note: `Ufvk` and `zcash_address::kind::unified::fvk::Ufvk` have similar names, but are actually distinct types
note: `Ufvk` is defined in crate `zcash_address`
   --> /home/alfredo/.cargo/registry/src/index.crates.io-6f17d22bba15001f/zcash_address-0.5.0/src/kind/unified/fvk.rs:106:1
    |
106 | pub struct Ufvk(pub(crate) Vec<Fvk>);
    | ^^^^^^^^^^^^^^^
note: `zcash_address::kind::unified::fvk::Ufvk` is defined in crate `zcash_address`
   --> /home/alfredo/.cargo/git/checkouts/librustzcash-0a74bd38a00f78b0/a1047ad/components/zcash_address/src/kind/unified/fvk.rs:106:1
    |
106 | pub struct Ufvk(pub(crate) Vec<Fvk>);
    | ^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `zcash_address` are being used?
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -116,5 +116,4 @@ zcash_protocol = { git = "https://github.com/zcash/librustzcash.git", rev = "141
 sapling-crypto = { git = "https://github.com/zcash/sapling-crypto", rev = "b1ad3694ee13a2fc5d291ad04721a6252da0993c" }
 orchard = { git = "https://github.com/zcash/orchard", rev = "55fb089a335bbbc1cda186c706bc037073df8eb7" }
 incrementalmerkletree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "ffe4234788fd22662b937ba7c6ea01535fcc1293" }
-incrementalmerkletree-testing = { git = "https://github.com/zcash/incrementalmerkletree", rev = "ffe4234788fd22662b937ba7c6ea01535fcc1293" }
 shardtree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "ffe4234788fd22662b937ba7c6ea01535fcc1293" }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -23,16 +23,16 @@ resolver = "2"
 
 [workspace.dependencies]
 incrementalmerkletree = "0.7.0"
-orchard = "0.10.0"
-sapling-crypto = "0.3.0"
-zcash_address = "0.6.0"
-zcash_client_backend = "0.14.0"
+orchard = "0.9.0"
+sapling-crypto = "0.2.0"
+zcash_address = "0.5.0"
+zcash_client_backend = "0.13.0"
 zcash_encoding = "0.2.1"
 zcash_history = "0.4.0"
-zcash_keys = "0.4.0"
-zcash_primitives = "0.19.0"
-zcash_proofs = "0.19.0"
-zcash_protocol = "0.4.0"
+zcash_keys = "0.3.0"
+zcash_primitives = "0.17.0"
+zcash_proofs = "0.17.0"
+zcash_protocol = "0.3.0"
 
 [workspace.metadata.release]
 
@@ -102,3 +102,15 @@ panic = "abort"
 #     - add "-flto=thin" to all C/C++ code builds
 #     - see https://doc.rust-lang.org/rustc/linker-plugin-lto.html#cc-code-as-a-dependency-in-rust
 lto = "thin"
+
+[patch.crates-io]
+incrementalmerkletree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "ffe4234788fd22662b937ba7c6ea01535fcc1293" }
+orchard = { git = "https://github.com/zcash/orchard", rev = "55fb089a335bbbc1cda186c706bc037073df8eb7" }
+sapling-crypto = { git = "https://github.com/zcash/sapling-crypto", rev = "b1ad3694ee13a2fc5d291ad04721a6252da0993c" }
+zcash_address = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
+zcash_client_backend = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
+zcash_encoding = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
+zcash_history = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
+zcash_primitives = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
+zcash_proofs = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
+zcash_protocol = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -103,10 +103,9 @@ panic = "abort"
 #     - see https://doc.rust-lang.org/rustc/linker-plugin-lto.html#cc-code-as-a-dependency-in-rust
 lto = "thin"
 
+# We can remove this patches after we get out of 2.0 release candidate and upgrade the ECC dpeendencies above.
+# This revisions are at the commit just before setting mainnet activation heights.
 [patch.crates-io]
-incrementalmerkletree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "ffe4234788fd22662b937ba7c6ea01535fcc1293" }
-orchard = { git = "https://github.com/zcash/orchard", rev = "55fb089a335bbbc1cda186c706bc037073df8eb7" }
-sapling-crypto = { git = "https://github.com/zcash/sapling-crypto", rev = "b1ad3694ee13a2fc5d291ad04721a6252da0993c" }
 zcash_address = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
 zcash_client_backend = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
 zcash_encoding = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
@@ -114,3 +113,8 @@ zcash_history = { git = "https://github.com/zcash/librustzcash.git", rev = "1410
 zcash_primitives = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
 zcash_proofs = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
 zcash_protocol = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }
+sapling-crypto = { git = "https://github.com/zcash/sapling-crypto", rev = "b1ad3694ee13a2fc5d291ad04721a6252da0993c" }
+orchard = { git = "https://github.com/zcash/orchard", rev = "55fb089a335bbbc1cda186c706bc037073df8eb7" }
+incrementalmerkletree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "ffe4234788fd22662b937ba7c6ea01535fcc1293" }
+incrementalmerkletree-testing = { git = "https://github.com/zcash/incrementalmerkletree", rev = "ffe4234788fd22662b937ba7c6ea01535fcc1293" }
+shardtree = { git = "https://github.com/zcash/incrementalmerkletree", rev = "ffe4234788fd22662b937ba7c6ea01535fcc1293" }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,17 +22,17 @@ resolver = "2"
 # `cargo release` settings
 
 [workspace.dependencies]
-incrementalmerkletree = "0.6.0"
-orchard = "0.9.0"
-sapling-crypto = "0.2.0"
-zcash_address = "0.5.0"
-zcash_client_backend = "0.13.0"
+incrementalmerkletree = "0.7.0"
+orchard = "0.10.0"
+sapling-crypto = "0.3.0"
+zcash_address = "0.6.0"
+zcash_client_backend = "0.14.0"
 zcash_encoding = "0.2.1"
 zcash_history = "0.4.0"
-zcash_keys = "0.3.0"
-zcash_primitives = "0.17.0"
-zcash_proofs = "0.17.0"
-zcash_protocol = "0.3.0"
+zcash_keys = "0.4.0"
+zcash_primitives = "0.19.0"
+zcash_proofs = "0.19.0"
+zcash_protocol = "0.4.0"
 
 [workspace.metadata.release]
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -103,7 +103,7 @@ panic = "abort"
 #     - see https://doc.rust-lang.org/rustc/linker-plugin-lto.html#cc-code-as-a-dependency-in-rust
 lto = "thin"
 
-# We can remove this patches after we get out of 2.0 release candidate and upgrade the ECC dpeendencies above.
+# We can remove this patches after we get out of 2.0 release candidate and upgrade the ECC dependencies above.
 # This revisions are at the commit just before setting mainnet activation heights.
 [patch.crates-io]
 zcash_address = { git = "https://github.com/zcash/librustzcash.git", rev = "1410f1449100a417bfbc4f6c7167aa9808e38792" }

--- a/deny.toml
+++ b/deny.toml
@@ -124,8 +124,7 @@ allow-git = [
     "https://github.com/zcash/librustzcash.git",
     "https://github.com/zcash/incrementalmerkletree",
     "https://github.com/zcash/orchard",
-    "https://github.com/zcash/sapling-crypto",
-    "https://github.com/zcash/incrementalmerkletree"
+    "https://github.com/zcash/sapling-crypto"
 ]
 
 [sources.allow-org]

--- a/deny.toml
+++ b/deny.toml
@@ -94,7 +94,16 @@ skip-tree = [
     { name = "uint", version = "=0.9.5" },
 
     # wait for dirs-sys to update windows-sys
-    { name = "windows-sys", version = "=0.48.0" }
+    { name = "windows-sys", version = "=0.48.0" },
+
+    # Remove after release candicate period is over and the ECC crates are not patched anymore
+    { name = "equihash", version = "=0.2.0" },
+    { name = "f4jumble", version = "=0.1.0" },
+    { name = "incrementalmerkletree", version = "=0.6.0" },
+    { name = "zcash_address", version = "=0.4.0" },
+    { name = "zcash_keys", version = "=0.3.0" },
+    { name = "zcash_primitives", version = "=0.16.0" },
+    { name = "zcash_protocol", version = "=0.2.0" }
 ]
 
 # This section is considered when running `cargo deny check sources`.

--- a/deny.toml
+++ b/deny.toml
@@ -71,9 +71,6 @@ skip-tree = [
     # also wait for ron to update insta, and wait for tonic update.
     { name = "base64", version = "=0.13.1" },
 
-    # wait for elasticsearch to update base64, darling, rustc_version, serde_with
-    { name = "elasticsearch", version = "=8.5.0-alpha.1" },
-
     # wait for reqwest to update base64
     { name = "base64", version = "=0.21.7" },
     { name = "sync_wrapper", version = "0.1.2" },

--- a/deny.toml
+++ b/deny.toml
@@ -121,7 +121,11 @@ unknown-git = "deny"
 allow-registry = ["https://github.com/rust-lang/crates.io-index"]
 # List of URLs for allowed Git repositories
 allow-git = [
-    "https://github.com/zcash/librustzcash.git"
+    "https://github.com/zcash/librustzcash.git",
+    "https://github.com/zcash/incrementalmerkletree",
+    "https://github.com/zcash/orchard",
+    "https://github.com/zcash/sapling-crypto",
+    "https://github.com/zcash/incrementalmerkletree"
 ]
 
 [sources.allow-org]

--- a/deny.toml
+++ b/deny.toml
@@ -97,6 +97,12 @@ skip-tree = [
 
     # wait for librocksdb-sys to update bindgen to one that uses newer itertools
     { name = "itertools", version = "=0.12.1" }
+
+    # wait for halo2_gadgets and primitive-types to update uint
+    { name = "uint", version = "=0.9.5" }
+
+    # wait for dirs-sys to update windows-sys
+    { name = "windows-sys", version = "=0.48.0" }
 ]
 
 # This section is considered when running `cargo deny check sources`.

--- a/deny.toml
+++ b/deny.toml
@@ -86,7 +86,7 @@ skip-tree = [
 
     # wait for structopt-derive to update heck
     { name = "heck", version = "=0.3.3" },
-,
+
     # wait for librocksdb-sys to update bindgen to one that uses newer itertools
     { name = "itertools", version = "=0.12.1" },
 

--- a/deny.toml
+++ b/deny.toml
@@ -84,22 +84,14 @@ skip-tree = [
     { name = "http-body", version = "=0.4.6" },
     { name = "hyper", version = "=0.14.30" },
 
-    # TODO: Remove this after we upgrade ECC dependencies to a crates.io version (#8749)
-    { name = "equihash", version = "=0.2.0" },
-    { name = "f4jumble", version = "=0.1.0" },
-    { name = "secp256k1", version = "=0.26.0" },
-    { name = "zcash_address", version = "=0.5.0" },
-    { name = "zcash_client_backend", version = "=0.13.0" },
-
-
     # wait for structopt-derive to update heck
     { name = "heck", version = "=0.3.3" },
-
+,
     # wait for librocksdb-sys to update bindgen to one that uses newer itertools
-    { name = "itertools", version = "=0.12.1" }
+    { name = "itertools", version = "=0.12.1" },
 
     # wait for halo2_gadgets and primitive-types to update uint
-    { name = "uint", version = "=0.9.5" }
+    { name = "uint", version = "=0.9.5" },
 
     # wait for dirs-sys to update windows-sys
     { name = "windows-sys", version = "=0.48.0" }

--- a/deny.toml
+++ b/deny.toml
@@ -71,6 +71,9 @@ skip-tree = [
     # also wait for ron to update insta, and wait for tonic update.
     { name = "base64", version = "=0.13.1" },
 
+    # wait for elasticsearch to update base64, darling, rustc_version, serde_with
+    { name = "elasticsearch", version = "=8.5.0-alpha.1" },
+
     # wait for reqwest to update base64
     { name = "base64", version = "=0.21.7" },
     { name = "sync_wrapper", version = "0.1.2" },

--- a/supply-chain/audits.toml
+++ b/supply-chain/audits.toml
@@ -1,6 +1,11 @@
 
 # cargo-vet audits file
 
+[audits]
+equihash = []
+f4jumble = []
+zcash_encoding = []
+
 [[audits.anstyle]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
@@ -25,6 +30,11 @@ delta = "0.5.1 -> 0.5.2"
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.4.0 -> 0.5.0"
+
+[[audits.bridgetree]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.5.0 -> 0.6.0"
 
 [[audits.bytemuck]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -91,18 +101,6 @@ who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.3.0 -> 0.4.0"
 
-[[audits.equihash]]
-who = "Alfredo Garcia <oxarbitrage@gmail.com>"
-criteria = "safe-to-deploy"
-delta = "0.2.0 -> 0.2.0@git:a1047adf0b6f324dad415db34762dc26f8367ce4"
-importable = false
-
-[[audits.f4jumble]]
-who = "Alfredo Garcia <oxarbitrage@gmail.com>"
-criteria = "safe-to-deploy"
-delta = "0.1.0 -> 0.1.0@git:a1047adf0b6f324dad415db34762dc26f8367ce4"
-importable = false
-
 [[audits.git2]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
@@ -127,6 +125,11 @@ delta = "0.1.6 -> 0.1.9"
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.5.1 -> 0.6.0"
+
+[[audits.incrementalmerkletree]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.6.0 -> 0.7.0"
 
 [[audits.indexmap]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -208,6 +211,11 @@ who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.8.0 -> 0.9.0"
 
+[[audits.orchard]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.9.0 -> 0.10.0"
+
 [[audits.owo-colors]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
@@ -283,6 +291,11 @@ who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.1.3 -> 0.2.0"
 
+[[audits.sapling-crypto]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.2.0 -> 0.3.0"
+
 [[audits.serde_spanned]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
@@ -317,6 +330,11 @@ version = "0.0.12"
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.3.1 -> 0.4.0"
+
+[[audits.shardtree]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.4.0 -> 0.5.0"
 
 [[audits.tempfile]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -541,8 +559,7 @@ delta = "0.4.0 -> 0.5.0"
 [[audits.zcash_address]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
-delta = "0.5.0 -> 0.5.0@git:a1047adf0b6f324dad415db34762dc26f8367ce4"
-importable = false
+delta = "0.5.0 -> 0.6.0"
 
 [[audits.zcash_client_backend]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -552,20 +569,12 @@ delta = "0.12.1 -> 0.13.0"
 [[audits.zcash_client_backend]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
-delta = "0.13.0 -> 0.13.0@git:a1047adf0b6f324dad415db34762dc26f8367ce4"
-importable = false
-
-[[audits.zcash_encoding]]
-who = "Alfredo Garcia <oxarbitrage@gmail.com>"
-criteria = "safe-to-deploy"
-delta = "0.2.1 -> 0.2.1@git:a1047adf0b6f324dad415db34762dc26f8367ce4"
-importable = false
+delta = "0.13.0 -> 0.14.0"
 
 [[audits.zcash_keys]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
-delta = "0.3.0 -> 0.3.0@git:a1047adf0b6f324dad415db34762dc26f8367ce4"
-importable = false
+delta = "0.3.0 -> 0.4.0"
 
 [[audits.zcash_primitives]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -575,13 +584,17 @@ delta = "0.16.0 -> 0.17.0"
 [[audits.zcash_primitives]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
-delta = "0.17.0 -> 0.17.0@git:a1047adf0b6f324dad415db34762dc26f8367ce4"
-importable = false
+delta = "0.17.0 -> 0.19.0"
 
 [[audits.zcash_proofs]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.16.0 -> 0.17.0"
+
+[[audits.zcash_proofs]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.17.0 -> 0.19.0"
 
 [[audits.zcash_protocol]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -596,8 +609,7 @@ delta = "0.1.1 -> 0.3.0"
 [[audits.zcash_protocol]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
-delta = "0.3.0 -> 0.3.0@git:a1047adf0b6f324dad415db34762dc26f8367ce4"
-importable = false
+delta = "0.3.0 -> 0.4.0"
 
 [[audits.zebra-chain]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -662,8 +674,7 @@ version = "0.1.0"
 [[audits.zip321]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
-delta = "0.1.0 -> 0.1.0@git:a1047adf0b6f324dad415db34762dc26f8367ce4"
-importable = false
+delta = "0.1.0 -> 0.2.0"
 
 [[trusted.clap]]
 criteria = "safe-to-deploy"

--- a/supply-chain/audits.toml
+++ b/supply-chain/audits.toml
@@ -1,11 +1,6 @@
 
 # cargo-vet audits file
 
-[audits]
-equihash = []
-f4jumble = []
-zcash_encoding = []
-
 [[audits.anstyle]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
@@ -101,6 +96,18 @@ who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.3.0 -> 0.4.0"
 
+[[audits.equihash]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.2.0 -> 0.2.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"
+importable = false
+
+[[audits.f4jumble]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.1.0 -> 0.1.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"
+importable = false
+
 [[audits.git2]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
@@ -130,6 +137,12 @@ delta = "0.5.1 -> 0.6.0"
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.6.0 -> 0.7.0"
+
+[[audits.incrementalmerkletree]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.7.0 -> 0.7.0@git:ffe4234788fd22662b937ba7c6ea01535fcc1293"
+importable = false
 
 [[audits.indexmap]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -216,6 +229,12 @@ who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.9.0 -> 0.10.0"
 
+[[audits.orchard]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.10.0 -> 0.9.1@git:55fb089a335bbbc1cda186c706bc037073df8eb7"
+importable = false
+
 [[audits.owo-colors]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
@@ -296,6 +315,12 @@ who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.2.0 -> 0.3.0"
 
+[[audits.sapling-crypto]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.3.0 -> 0.2.0@git:b1ad3694ee13a2fc5d291ad04721a6252da0993c"
+importable = false
+
 [[audits.serde_spanned]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
@@ -335,6 +360,12 @@ delta = "0.3.1 -> 0.4.0"
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.4.0 -> 0.5.0"
+
+[[audits.shardtree]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.5.0 -> 0.4.0@git:ffe4234788fd22662b937ba7c6ea01535fcc1293"
+importable = false
 
 [[audits.tempfile]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -559,6 +590,12 @@ delta = "0.4.0 -> 0.5.0"
 [[audits.zcash_address]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
+delta = "0.5.0 -> 0.5.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"
+importable = false
+
+[[audits.zcash_address]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
 delta = "0.5.0 -> 0.6.0"
 
 [[audits.zcash_client_backend]]
@@ -571,15 +608,45 @@ who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.13.0 -> 0.14.0"
 
+[[audits.zcash_client_backend]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.14.0 -> 0.13.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"
+importable = false
+
+[[audits.zcash_encoding]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.2.1 -> 0.2.1@git:1410f1449100a417bfbc4f6c7167aa9808e38792"
+importable = false
+
+[[audits.zcash_history]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.4.0 -> 0.4.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"
+importable = false
+
 [[audits.zcash_keys]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.3.0 -> 0.4.0"
 
+[[audits.zcash_keys]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.4.0 -> 0.3.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"
+importable = false
+
 [[audits.zcash_primitives]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.16.0 -> 0.17.0"
+
+[[audits.zcash_primitives]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.17.0 -> 0.17.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"
+importable = false
 
 [[audits.zcash_primitives]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -590,6 +657,12 @@ delta = "0.17.0 -> 0.19.0"
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.16.0 -> 0.17.0"
+
+[[audits.zcash_proofs]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.17.0 -> 0.17.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"
+importable = false
 
 [[audits.zcash_proofs]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -610,6 +683,12 @@ delta = "0.1.1 -> 0.3.0"
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.3.0 -> 0.4.0"
+
+[[audits.zcash_protocol]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.4.0 -> 0.3.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"
+importable = false
 
 [[audits.zebra-chain]]
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
@@ -675,6 +754,12 @@ version = "0.1.0"
 who = "Alfredo Garcia <oxarbitrage@gmail.com>"
 criteria = "safe-to-deploy"
 delta = "0.1.0 -> 0.2.0"
+
+[[audits.zip321]]
+who = "Alfredo Garcia <oxarbitrage@gmail.com>"
+criteria = "safe-to-deploy"
+delta = "0.2.0 -> 0.1.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"
+importable = false
 
 [[trusted.clap]]
 criteria = "safe-to-deploy"

--- a/supply-chain/config.toml
+++ b/supply-chain/config.toml
@@ -16,10 +16,66 @@ url = "https://raw.githubusercontent.com/zcash/rust-ecosystem/main/supply-chain/
 [imports.zcashd]
 url = "https://raw.githubusercontent.com/zcash/zcash/master/qa/supply-chain/audits.toml"
 
+[policy."equihash:0.2.0"]
+
+[policy."equihash:0.2.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"]
+audit-as-crates-io = true
+
+[policy."f4jumble:0.1.0"]
+
+[policy."f4jumble:0.1.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"]
+audit-as-crates-io = true
+
+[policy."incrementalmerkletree:0.6.0"]
+
+[policy."incrementalmerkletree:0.7.0@git:ffe4234788fd22662b937ba7c6ea01535fcc1293"]
+audit-as-crates-io = true
+
+[policy.orchard]
+audit-as-crates-io = true
+
+[policy.sapling-crypto]
+audit-as-crates-io = true
+
+[policy.shardtree]
+audit-as-crates-io = true
+
 [policy.tower-batch-control]
 audit-as-crates-io = true
 
 [policy.tower-fallback]
+audit-as-crates-io = true
+
+[policy."zcash_address:0.4.0"]
+
+[policy."zcash_address:0.5.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"]
+audit-as-crates-io = true
+
+[policy.zcash_client_backend]
+audit-as-crates-io = true
+
+[policy.zcash_encoding]
+audit-as-crates-io = true
+
+[policy.zcash_history]
+audit-as-crates-io = true
+
+[policy."zcash_keys:0.3.0"]
+
+[policy."zcash_keys:0.3.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"]
+audit-as-crates-io = true
+
+[policy."zcash_primitives:0.16.0"]
+
+[policy."zcash_primitives:0.17.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"]
+audit-as-crates-io = true
+
+[policy.zcash_proofs]
+audit-as-crates-io = true
+
+[policy."zcash_protocol:0.2.0"]
+
+[policy."zcash_protocol:0.3.0@git:1410f1449100a417bfbc4f6c7167aa9808e38792"]
 audit-as-crates-io = true
 
 [policy.zebra-chain]
@@ -56,6 +112,9 @@ audit-as-crates-io = true
 audit-as-crates-io = true
 
 [policy.zebrad]
+audit-as-crates-io = true
+
+[policy.zip321]
 audit-as-crates-io = true
 
 [[exemptions.abscissa_core]]

--- a/supply-chain/config.toml
+++ b/supply-chain/config.toml
@@ -16,34 +16,10 @@ url = "https://raw.githubusercontent.com/zcash/rust-ecosystem/main/supply-chain/
 [imports.zcashd]
 url = "https://raw.githubusercontent.com/zcash/zcash/master/qa/supply-chain/audits.toml"
 
-[policy.equihash]
-audit-as-crates-io = true
-
-[policy.f4jumble]
-audit-as-crates-io = true
-
 [policy.tower-batch-control]
 audit-as-crates-io = true
 
 [policy.tower-fallback]
-audit-as-crates-io = true
-
-[policy.zcash_address]
-audit-as-crates-io = true
-
-[policy.zcash_client_backend]
-audit-as-crates-io = true
-
-[policy.zcash_encoding]
-audit-as-crates-io = true
-
-[policy.zcash_keys]
-audit-as-crates-io = true
-
-[policy.zcash_primitives]
-audit-as-crates-io = true
-
-[policy.zcash_protocol]
 audit-as-crates-io = true
 
 [policy.zebra-chain]
@@ -80,9 +56,6 @@ audit-as-crates-io = true
 audit-as-crates-io = true
 
 [policy.zebrad]
-audit-as-crates-io = true
-
-[policy.zip321]
 audit-as-crates-io = true
 
 [[exemptions.abscissa_core]]

--- a/zebra-chain/Cargo.toml
+++ b/zebra-chain/Cargo.toml
@@ -88,7 +88,7 @@ primitive-types = "0.12.2"
 rand_core = "0.6.4"
 ripemd = "0.1.3"
 # Matches version used by hdwallet
-secp256k1 = { version = "0.26.0", features = ["serde"] }
+secp256k1 = { version = "0.27.0", features = ["serde"] }
 sha2 = { version = "0.10.7", features = ["compress"] }
 uint = "0.10.0"
 x25519-dalek = { version = "2.0.1", features = ["serde"] }

--- a/zebra-chain/Cargo.toml
+++ b/zebra-chain/Cargo.toml
@@ -68,7 +68,7 @@ bitflags = "2.5.0"
 bitflags-serde-legacy = "0.1.1"
 blake2b_simd = "1.0.2"
 blake2s_simd = "1.0.2"
-bridgetree = "0.5.0"
+bridgetree = "0.6.0"
 bs58 = { version = "0.5.1", features = ["check"] }
 byteorder = "1.5.0"
 

--- a/zebra-scan/Cargo.toml
+++ b/zebra-scan/Cargo.toml
@@ -71,11 +71,10 @@ tracing = "0.1.39"
 futures = "0.3.30"
 
 # ECC dependencies.
-# TODO: we can't use the workspace version for all ECC dependencies in this crate yet (#8809)
-zcash_client_backend = { git = "https://github.com/zcash/librustzcash/", commit = "40ca428c6081c61d5a2bf3f2053eb9e18219ca95" }
+zcash_client_backend.workspace = true
 zcash_keys = { workspace = true, features = ["sapling"] }
-zcash_primitives = { git = "https://github.com/zcash/librustzcash/", commit = "40ca428c6081c61d5a2bf3f2053eb9e18219ca95" }
-zcash_address = { git = "https://github.com/zcash/librustzcash/", commit = "40ca428c6081c61d5a2bf3f2053eb9e18219ca95" }
+zcash_primitives.workspace = true
+zcash_address.workspace = true
 sapling-crypto.workspace = true
 
 zebra-chain = { path = "../zebra-chain", version = "1.0.0-beta.39", features = ["shielded-scan"] }

